src/NLog/Contexts/ScopeContext.cs

```diff
@@ -0,0 +1,994 @@
+﻿// 
+// Copyright (c) 2004-2020 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System;
+using System.Collections.Generic;
+using NLog.Internal;
+
+namespace NLog
+{
+    /// <summary>
+    /// <see cref="ScopeContext"/> allows one to store state in the thread execution context. All LogEvents created
+    /// within a scope can include the scope state when wanted. The logical operation scope state supports both
+    /// scope-properties and scope-operation-states.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="ScopeContext"/> unifies <see cref="MappedDiagnosticsLogicalContext"/> (MDLC) and <see cref="NestedDiagnosticsLogicalContext"/> (NDLC).
+    /// 
+    /// .NetCore (and .Net46) uses AsyncLocal for handling the thread execution context. Older .NetFramework uses System.Runtime.Remoting.CallContext
+    /// </remarks>
+    public static class ScopeContext
+    {
+#if !NET35 && !NET40
+        /// <summary>
+        /// Creates logical operation scope that includes provided properties and stack-value
+        /// </summary>
+        /// <param name="operationState">Value to added to the scope operation stack</param>
+        /// <param name="properties">Properties being added to the scope dictionray</param>
```

Comment:
dictionray ;)

Reply:
Thanks. Fixed spelling. But Azure-build is now in cancelled state. Can you give it a kick? (Instead of me force-pushing)